src/main/java/teammates/common/util/Config.java

```diff
@@ -103,53 +98,113 @@ public final class Config {
-        MAILJET_APIKEY = properties.getProperty("app.mailjet.apikey");
-        MAILJET_SECRETKEY = properties.getProperty("app.mailjet.secretkey");
-        SEARCH_SERVICE_HOST = properties.getProperty("app.search.service.host");
-        ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(properties.getProperty("app.enable.datastore.backup", "false"));
-        MAINTENANCE = Boolean.parseBoolean(properties.getProperty("app.maintenance", "false"));
+                Arrays.asList(getProperty(properties, devProperties, "app.maintainers", "").split(",")));
+        SUPPORT_EMAIL = getProperty(properties, devProperties, "app.crashreport.email");
+        EMAIL_SENDEREMAIL = getProperty(properties, devProperties, "app.email.senderemail");
+        EMAIL_SENDERNAME = getProperty(properties, devProperties, "app.email.sendername");
+        EMAIL_REPLYTO = getProperty(properties, devProperties, "app.email.replyto");
+        EMAIL_SERVICE = getProperty(properties, devProperties, "app.email.service");
+        SENDGRID_APIKEY = getProperty(properties, devProperties, "app.sendgrid.apikey");
+        MAILGUN_APIKEY = getProperty(properties, devProperties, "app.mailgun.apikey");
+        MAILGUN_DOMAINNAME = getProperty(properties, devProperties, "app.mailgun.domainname");
+        MAILJET_APIKEY = getProperty(properties, devProperties, "app.mailjet.apikey");
+        MAILJET_SECRETKEY = getProperty(properties, devProperties, "app.mailjet.secretkey");
+        SEARCH_SERVICE_HOST = getProperty(properties, devProperties, "app.search.service.host");
+        ENABLE_DATASTORE_BACKUP = Boolean.parseBoolean(
+                getProperty(properties, devProperties, "app.enable.datastore.backup", "false"));
+        MAINTENANCE = Boolean.parseBoolean(getProperty(properties, devProperties, "app.maintenance", "false"));
+
+        // Dev property values, read directly from devProperties
+        APP_LOCALDATASTORE_PORT = Integer.parseInt(
+                devProperties.getProperty("app.localdatastore.port", "8484"));
+        ENABLE_DEVSERVER_LOGIN = Boolean.parseBoolean(
+                devProperties.getProperty("app.enable.devserver.login", "true"));
+        TASKQUEUE_ACTIVE = Boolean.parseBoolean(
+                devProperties.getProperty("app.taskqueue.active", "true"));
+        APP_BACKEND_URL = getProperty(properties, devProperties, "app.backend.url", APP_FRONTEND_URL);
     }
 
     private Config() {
         // access static fields directly
     }
 
-    static String getBaseAppUrl() {
-        return isDevServer() ? "http://localhost:" + getPort() : "https://" + APP_ID + ".appspot.com";
+    private static String getProperty(Properties properties, Properties devProperties, String key, String defaultValue) {
+        if (IS_DEV_SERVER) {
+            return devProperties.getProperty(key, properties.getProperty(key, defaultValue));
+        } else {
+            return properties.getProperty(key, defaultValue);
+        }
```

Comment:
```java
if (IS_DEV_SERVER && devProperties.getProperty(key) != null) {
    return devProperties.getProperty(key);
}
return defaultValue == null ? properties.getProperty(key) : properties.getProperty(key, defaultValue);
```
Then the second method without `defaultValue` can just call this, no repetition needed.